### PR TITLE
[feat]: [CDS-35390]: Delete file or folder

### PR DESCRIPTION
**Pull request by vladimir-peric**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32066)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/FileStoreResource.java [hunk 1]
```diff
@@ -91,4 +93,26 @@
     return ResponseDTO.newResponse(
         fileStoreService.listFolderNodes(accountIdentifier, orgIdentifier, projectIdentifier, folderNodeDTO));
   }
+
+  @DELETE
+  @Consumes({"application/json"})
+  @Path("folder")
+  @ApiOperation(value = "Delete file or folder by identifier", nickname = "deleteFile")
+  @Operation(operationId = "getFile", summary = "Delete file or folder by identifier",
```
Comment (filip5rovic) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/FileStoreResource.java:101]: I guess that `operationId` should be `deleteFile`, right?

File: 120-ng-manager/src/main/java/io/harness/ng/core/remote/FileStoreResource.java [hunk 2]
```diff
@@ -91,4 +93,26 @@
     return ResponseDTO.newResponse(
         fileStoreService.listFolderNodes(accountIdentifier, orgIdentifier, projectIdentifier, folderNodeDTO));
   }
+
+  @DELETE
+  @Consumes({"application/json"})
+  @Path("folder")
```
Comment (filip5rovic) [inline 120-ng-manager/src/main/java/io/harness/ng/core/remote/FileStoreResource.java:99]: This path also looks suspicious :suspect:

**Comment by vladimir-peric** [comment]
*[120-ng-manager/src/main/java/io/harness/ng/core/remote/FileStoreResource.java, line 99 at r1](https://reviewable.io/reviews/harness/harness-core/32066#-MzvzAcZWDpVatk_aehc-r1-99:-MzyjFOW69zqIJZp6klR:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2f9e109615d7b85c1f9bf74e2f2d2d04ea02acc6/120-ng-manager/src/main/java/io/harness/ng/core/remote/FileStoreResource.java#L99)):*
<details><summary><i>Previously, filip5rovic (Filip Petrovic) wrote…</i></summary><blockquote>

This path also looks suspicious :suspect: 
</blockquote></details>

Done.

**Review by vladimir-peric — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32066)* status: 5 file(s) to review. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[120-ng-manager/src/main/java/io/harness/ng/core/remote/FileStoreResource.java, line 101 at r1](https://reviewable.io/reviews/harness/harness-core/32066#-MzvzAcZWDpVatk_aehc-r1-101:-MzyjEHeBDdt7MMKaMgf:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/2f9e109615d7b85c1f9bf74e2f2d2d04ea02acc6/120-ng-manager/src/main/java/io/harness/ng/core/remote/FileStoreResource.java#L101)):*
<details><summary><i>Previously, filip5rovic (Filip Petrovic) wrote…</i></summary><blockquote>

I guess that `operationId` should be `deleteFile`, right?
</blockquote></details>

Done.

**Review by ivan-83 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32066#-:-MzzYzrwCYAXoTAv_q_y:bnfp4nl)* status: 5 file(s) to review. 2. unresolved discussions. 

___
*[120-ng-manager/src/main/java/io/harness/ng/core/api/impl/FileStoreServiceImpl.java, line 164 at r3](https://reviewable.io/reviews/harness/harness-core/32066#-MzzYO4UDPP61WQqdCcE:-MzzYO4V3HOqDGjB5ycA:b33lkdf) ([raw file](https://github.com/harness/harness-core/blob/e15745d0201acf6809df9edb789ce677e61d7203/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/FileStoreServiceImpl.java#L164)):*
> ```Java
>     try {
>       fileStoreRepository.delete(file);
>       fileService.deleteFile(file.getEntityId(), CONFIGS);
> ```

First  fileService.deleteFile(file.getEntityId(), CONFIGS); and after that  fileStoreRepository.delete(file);



_Code quote:_
```Java
private boolean deleteFile(NGFile file)
```
___
*[120-ng-manager/src/main/java/io/harness/ng/core/api/impl/FileStoreServiceImpl.java, line 191 at r3](https://reviewable.io/reviews/harness/harness-core/32066#-MzzYkN0BqTLoInodfS5:-MzzYkN0BqTLoInodfS6:b-sfsulp) ([raw file](https://github.com/harness/harness-core/blob/e15745d0201acf6809df9edb789ce677e61d7203/120-ng-manager/src/main/java/io/harness/ng/core/api/impl/FileStoreServiceImpl.java#L191)):*
> ```Java
>   }
> 
>   private boolean deleteFileOrFolder(NGFile fileOrFolder) {
> ```

Minor reorder methods in class how it is invoked.

**Pull request merged**